MyJournal/MyJournal/View Controllers/ManageEntryVC.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  ManageEntryVC:.swift
+//  MyJournal
+//
+//  Created by Jeffrey Santana on 8/19/19.
+//  Copyright © 2019 Lambda. All rights reserved.
+//
+
+import UIKit
+
+class ManageEntryVC: UIViewController {
+
+	//MARK: - IBOutlets
+	
+	@IBOutlet weak var titleTextField: UITextField!
+	@IBOutlet weak var storyTextView: UITextView!
+	
+	//MARK: - Properties
+	
+	var journalController: JournalController!
+	var entry: Entry?
+	
+	//MARK: - Life Cycle
+	
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		
+		updateViews()
+	}
+	
+	//MARK: - IBActions
+	
+	@IBAction func saveBtnTapped(_ sender: Any) {
+		guard let title = titleTextField.optionalText else { return }
+		let story  = storyTextView.text
+		
+		if let entry = entry {
+			journalController.updateEntry(entry: entry, title: title, story: story)
+		} else {
+			journalController.createEntry(title: title, story: story)
+		}
+		
+		okAction(message: entry == nil ? "New journal entry created." : "Journal entry updated.")
+	}
+	
+	//MARK: - Helpers
+	
+	private func updateViews() {
+		guard let entry = entry else { return }
+		
+		titleTextField.text = entry.title
+		storyTextView.text = entry.story
+	}
+	
```

Comment:
I would personally argue that this popup is unnecessary as it adds friction to the user experience.